org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/handlers/ProgressReporterManager.java

```diff
@@ -169,6 +169,7 @@ private class ProgressReporter extends CancellableProgressMonitor {
 		protected long lastReport = 0;
 		protected String progressId;
 		private boolean sentBegin = false;
+		private boolean sentEnd = false;
```

Comment:
I would just add a comment for this field to remind us the reason we have it, is because we observed some jobs (eg. "Synchronizing projects") reporting the same (end) progress twice. For Eclipse itself, this is not wrong, but we try to avoid this by only marking the final progress as "done".

Reply:
"some tasks were reporting duplicate end messages, so this ensures only one is sent to client"